pkg/stargate/deployments_test.go

```diff
@@ -463,7 +467,8 @@ func testNewDeploymentsManyRacksCustomAffinityStargate(t *testing.T) {
 	assert.Contains(t, deployment3.Spec.Template.Labels, api.StargateDeploymentLabel)
 	assert.Equal(t, "cluster1-dc1-rack3-stargate-deployment", deployment3.Spec.Template.Labels[api.StargateDeploymentLabel])
 	assert.Equal(t, stargate.Spec.Racks[0].Affinity, deployment3.Spec.Template.Spec.Affinity)
-	assert.Equal(t, stargate.Spec.Racks[0].NodeSelector, deployment3.Spec.Template.Spec.NodeSelector)
+	// node selectors should have been merged
+	assert.Equal(t, map[string]string{"selectorKey2": "selectorValue2", "selectorKey3": "selectorValue3"}, deployment3.Spec.Template.Spec.NodeSelector)
```

Comment:
Is this a test for the piece of the merge logic which was broken? It seems like a new test...

Reply:
This is the only place where the effects of switching to a true merging behavior is visible in our tests.

Previously, the test was expecting the resulting selectors map to be equal to the map in the rack template, effectively ignoring the map in the DC template.

With the merge, the two maps are being merged and combined together – which is imo the right thing to do.

Reply:
> With the merge, the two maps are being merged and combined together – which is imo the right thing to do.

Yes, I agree, the merge operation should be 'deep' (i.e. merge maps and arrays within the struct).